addons/repair/functions/fnc_addRepairActions.sqf

```diff
@@ -29,154 +29,123 @@ _initializedClasses = GETMVAR(GVAR(initializedClasses),[]);
+        // An action to replace the wheel is required
+        _name = format ["Replace_%1_%2", _forEachIndex, _selection];
         _text = localize LSTRING(ReplaceWheel);
-
         _condition = {[_this select 1, _this select 0, _this select 2 select 0, "ReplaceWheel"] call DFUNC(canRepair)};
         _statement = {[_this select 1, _this select 0, _this select 2 select 0, "ReplaceWheel"] call DFUNC(repair)};
-
-        _action = [_name, _text, _icon, _statement, _condition, {}, [_x], _selection, 2] call EFUNC(interact_menu,createAction);
+        _action = [_name, _text, _icon, _statement, _condition, {}, [_hitpoint], _selection, 2] call EFUNC(interact_menu,createAction);
         [_type, 0, [], _action] call EFUNC(interact_menu,addActionToClass);
-
     } else {
-        // exit if the hitpoint is in the blacklist, e.g. glasses
-        if (_x in IGNORED_HITPOINTS) exitWith {TRACE_3("Ignored Hitpoint",_x,_forEachIndex,_selectionName);};
-        if (_x == "") exitWith {TRACE_3("Hitpoint Empty",_x,_forEachIndex,_selectionName);};
-        if (isNil {_vehicle getHit _selectionName}) exitWith {TRACE_3("No Selection",_x,_forEachIndex,_selectionName);}; //Not a real hitpoint
-
-        private ["_hitpointGroupConfig", "_inHitpointSubGroup", "_currentHitpoint"];
-
-        // Get hitpoint groups if available
-        _hitpointGroupConfig = configFile >> "CfgVehicles" >> _type >> QGVAR(hitpointGroups);
-        _inHitpointSubGroup = false;
-        if (isArray _hitpointGroupConfig) then {
-            // Set variable if current hitpoint is in a sub-group (to be excluded from adding action)
-            _currentHitpoint = _x;
+        // Empty selections don't exist
+        // Empty hitpoints don't contain enough information
+        if (_selection isEqualTo "") exitWith {};
+        if (_hitpoint isEqualTo "") exitWith {};
+
+        // Associated hitpoints can be grouped via config to produce a single repair action
+        _groupsConfig = configFile >> "CfgVehicles" >> _type >> QGVAR(hitpointGroups);
+        if (isArray _groupsConfig) then {
+            _childHitPoint = false;
             {
                 {
-                    if (_x == _currentHitpoint) exitWith {
-                        _inHitpointSubGroup = true;
+                    if (_hitpoint == _x) exitWith {
+                        _childHitPoint = true;
                     };
                 } forEach (_x select 1);
-            } forEach (getArray _hitpointGroupConfig);
+            } forEach (getArray _groupsConfig);
         };
-
-        // Exit if current hitpoint is in sub-group (only main hitpoints get actions)
-        if (_inHitpointSubGroup) exitWith {TRACE_3("inHitpointSubGroup",_x,_forEachIndex,_selectionName);};
-
-        // exit if the hitpoint is virtual
-        if (isText (configFile >> "CfgVehicles" >> _type >> "HitPoints" >> _x >> "depends")) exitWith {TRACE_3("Depends",_x,_forEachIndex,_selectionName);};
```

Comment:
Not entirely, however if a selection exists in the model and has a respective hit point then I'm presuming it's not "virtual" since it exists in the model to being with. :stuck_out_tongue: